src/main/java/Deadline.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * The Deadline class is a type of Task.
+ */
+public class Deadline extends Task {
+
+    protected String by;
+
+    /**
+     * Creates a Deadline Task object.
+     *
+     * @param description A string representation of the task description.
+     * @param by A string representation of the deadline for the task.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
```

Comment:
Can consider giving by a more name that is more self-explanatory which can make readers easier to follow through.